test/gtest/uct/cuda/test_switch_cuda_device.cc

```diff
@@ -143,3 +144,68 @@ UCS_TEST_P(test_switch_cuda_device, detect_mem_type_cuda_fabric)
 #endif
 
 _UCT_MD_INSTANTIATE_TEST_CASE(test_switch_cuda_device, cuda_cpy);
+
+class test_p2p_create_destroy_ctx : public uct_p2p_rma_test {
+public:
+    void test_xfer(send_func_t send, size_t length, unsigned flags,
+                   ucs_memory_type_t mem_type) override;
+};
+
+void test_p2p_create_destroy_ctx::test_xfer(send_func_t send, size_t length,
+                                            unsigned flags,
+                                            ucs_memory_type_t mem_type)
+{
+    int num_devices;
+    ASSERT_EQ(cuDeviceGetCount(&num_devices), CUDA_SUCCESS);
+
+    if (num_devices < 1) {
+        UCS_TEST_SKIP_R("no cuda devices available");
+    }
+
+    CUcontext current_ctx;
+    ASSERT_EQ(cuCtxGetCurrent(&current_ctx), CUDA_SUCCESS);
+
+    CUdevice device;
+    if (current_ctx != NULL) {
+        ASSERT_EQ(cuCtxGetDevice(&device), CUDA_SUCCESS);
```

Comment:
why not always use cuDeviceGet(&device, 0)?
IMO better to not use/assume anything about context being set from previous calls, and always create/use primary context explicitly.